src/plugins/gui/providers/ioniaProvider.ts

```diff
@@ -392,34 +413,75 @@ export const makeIoniaProvider: FiatProviderFactory<IoniaMethods> = {
+        const fiatAmount = quoteParams.amountType === 'fiat' ? quoteParams.exchangeAmount : mul(quoteParams.exchangeAmount, toBigNumberString(price))
+
+        // Concurrently get the purchase card
+        const purchaseCardPromise = getPurchaseCard(quoteParams.displayCurrencyCode, parseFloat(fiatAmount))
 
         checkAmountMinMax(parseFloat(fiatAmount))
 
         const fiatProviderQuote: FiatProviderQuote = {
           providerId: makeIoniaProvider.providerId,
           partnerIcon: fiatProvider.partnerIcon,
           pluginDisplayName: '',
-          displayCurrencyCode: params.displayCurrencyCode,
+          displayCurrencyCode: quoteParams.displayCurrencyCode,
           cryptoAmount,
           isEstimate: true,
-          fiatCurrencyCode: params.fiatCurrencyCode,
+          fiatCurrencyCode: quoteParams.fiatCurrencyCode,
           fiatAmount,
           direction: 'sell',
-          regionCode: params.regionCode,
-          paymentTypes: params.paymentTypes,
-
-          approveQuote: async params => {},
+          regionCode: quoteParams.regionCode,
+          paymentTypes: quoteParams.paymentTypes,
+
+          approveQuote: async approveParams => {
+            const { showUi, coreWallet: wallet } = approveParams
+
+            // Unbox the latest purchase card:
+            const purchaseCard = await purchaseCardPromise
+
+            // Use the purchase card URI to complete the payment over the Payment Protocol:
+            const parsedUri: EdgeParsedUri & { paymentProtocolUrl?: string } = await wallet.parseUri(purchaseCard.uri, quoteParams.displayCurrencyCode)
+            const { paymentProtocolUrl } = parsedUri
+            if (paymentProtocolUrl == null) {
+              throw new Error(lstrings.missing_provider_payment_address_message)
+            }
+
+            // Check if the URI requires a warning to the user:
+            const approved = await addressWarnings(parsedUri, quoteParams.displayCurrencyCode)
+            if (!approved) return
+
+            console.log(`Show send of ${cryptoAmount} ${quoteParams.displayCurrencyCode} to '${purchaseCard.uri}' to purchase ${fiatAmount} USD card.`)
+
+            return await new Promise((resolve, reject) => {
+              showUi
+                .sendPaymentProto({
+                  uri: paymentProtocolUrl,
+                  params: {
+                    wallet: wallet,
+                    currencyCode: quoteParams.displayCurrencyCode,
+                    onBack: () => {
+                      reject(
+                        new Error(
+                          `User cancelled quote of ${cryptoAmount} ${quoteParams.displayCurrencyCode} to '${purchaseCard.uri}' to purchase ${fiatAmount} USD card.`
+                        )
+                      )
+                    },
+                    onDone: resolve
+                  }
+                })
+                .catch(reject)
```

Comment:
don't pass reject directly. Instead, put anonymous function that logs the error object and where you are in the code with some context tagging.  Then call the reject(error). This gives us some context.